code/datums/emergency_calls/emergency_call.dm

```diff
@@ -146,76 +146,92 @@
-				return
-
-			candidates = list() //Blank out the candidates list for next time.
-
-			spawn(COOLDOWN_COMM_REQUEST)
-				SSticker.mode.on_distress_cooldown = FALSE
+	addtimer(CALLBACK(src, .do_activate, announce), 1 MINUTES)
 
+/datum/emergency_call/proc/do_activate(announce = TRUE)
+	if(length(candidates) < mob_min)
+		message_admins("Aborting distress beacon [name], not enough candidates. Found: [length(candidates)]. Minimum required: [mob_min].")
+		SSticker.mode.waiting_for_candidates = FALSE
+		members = list() //Empty the members list.
+		candidates = list()
+
+		if(announce)
+			command_announcement.Announce("The distress signal has not received a response, the launch tubes are now recalibrating.", "Distress Beacon")
+
+		SSticker.mode.picked_call = null
+		SSticker.mode.on_distress_cooldown = TRUE
+
+		addtimer(CALLBACK(src, .distress_cooldown), COOLDOWN_COMM_REQUEST)
+		return
+
+	SSticker.mode.waiting_for_candidates = FALSE
+	var/datum/mind/picked_candidates = list()
+	if(mob_max > 0)
+		for(var/i = 1 to mob_max)
+			if(!length(candidates)) //We ran out of candidates.
+				break
+			var/datum/mind/M = pick(candidates) //Get a random candidate, then remove it from the candidates list.
+			if(!istype(M)) //Something went horrifically wrong
+				candidates -= M
+				continue
+			if(M.current?.stat != DEAD)
+				candidates -= M //Strip them from the list, they aren't dead anymore.
+				continue
+			if(name == "Xenomorphs" && is_banned_from(M.current.ckey, ROLE_XENOMORPH))
+				candidates -= M
+				continue
+			picked_candidates += M
+			candidates -= M
+
+		if(length(candidates))
+			for(var/datum/mind/M in candidates)
+				if(M.current)
+					to_chat(M.current, "<span class='warning'>You didn't get selected to join the distress team. Better luck next time!</span>")
+
+	if(announce)
+		command_announcement.Announce(dispatch_message, "Distress Beacon", new_sound='sound/AI/distressreceived.ogg') //Announcement that the Distress Beacon has been answered, does not hint towards the chosen ERT
+
+	message_admins("Distress beacon: [name] finalized, setting up candidates.")
+
+	// begin loading the shuttle
+	if(!SSmapping.shuttle_templates[shuttle_id])
+		CRASH("ert called with invalid shuttle_id")
+		return
+	var/datum/map_template/shuttle/S = SSmapping.shuttle_templates[shuttle_id]
+
+	var/obj/docking_port/stationary/L = SSshuttle.getDock("distress_loading")
+	if(!L)
+		CRASH("no distress loading port defined")
+
+	if(L.get_docked())
+		CRASH("trying to load an ert when one is currently being loaded")
+
+	shuttle = SSshuttle.action_load(S, L)
+
+	if(!istype(shuttle))
+		CRASH("ert shuttle failed to load")
+
+	spawn_items()
+
+	if(length(picked_candidates))
+		max_medics = max(round(length(members) * 0.25), 1)
+		for(var/datum/mind/M in picked_candidates)
+			members += M
+			create_member(M)
+	else
+		message_admins("ERROR: No picked candidates, aborting.")
+		return
+
+	if(auto_shuttle_launch)
```

Comment:
```suggestion
	if(auto_shuttle_launch && !shuttle.auto_launch())
```
And remove the following line and fix indentation?

Reply:
readability

Reply:
This doesn't really increase readability. On the contrary.

Reply:
Disagree